dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/PythonCommandExecutor.java

```diff
@@ -185,4 +190,34 @@ public static String getPythonHome(String envPath) {
         return null;
     }
 
+    /**
+     * get python home from the environment config
+     *
+     * @param environmentConfig env config
+     * @return python home
+     */
+    public static String getPythonHomeFromEnvironmentConfig(String environmentConfig) {
+        BufferedReader br = null;
+        StringBuilder sb = new StringBuilder();
+
+        String[] lines = environmentConfig.split("\n");
+
+        for (String line : lines) {
+            if (StringUtils.isNotEmpty(StringUtils.trim(line))) {
+                if (line.contains(Constants.PYTHON_HOME)) {
+                    sb.append(line);
```

Comment:
It's better to use string here.
Or you can just use stream.filter() to find the first one.

Reply:
done